rust/psibase/src/main.rs

```diff
@@ -967,6 +1017,108 @@ async fn search(
     Ok(())
 }
 
+struct ServicePrinter<'a> {
+    service: AccountNumber,
+    info: Option<&'a ServiceInfo>,
+    data: &'a [PackageDataFile],
+}
+
+impl fmt::Display for ServicePrinter<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let Some(info) = &self.info {
+            writeln!(f, "service: {}", self.service)?;
+            if !info.flags.is_empty() {
+                write!(f, "  flags:")?;
+                for flag in &info.flags {
+                    write!(f, " {}", flag)?;
+                }
+                writeln!(f, "")?;
+            }
+            if let Some(server) = &info.server {
+                writeln!(f, "  server: {}", server)?;
+            }
+        } else {
+            writeln!(f, "account: {}", self.service)?;
+        }
+        if !self.data.is_empty() {
+            writeln!(f, "  files:")?;
+            for file in self.data {
+                writeln!(f, "    {}", &file.filename)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+fn get_service_data(s: &[PackageDataFile], account: AccountNumber) -> &[PackageDataFile] {
+    let key = account.to_string();
+    let lower = s.partition_point(|file| &file.account.to_string() < &key);
+    let upper = s.partition_point(|file| &file.account.to_string() <= &key);
+    &s[lower..upper]
+}
+
+async fn show_package<T: PackageRegistry + ?Sized>(
+    reg: &T,
+    base_url: &reqwest::Url,
+    client: &mut reqwest::Client,
+    package: &Meta,
+    origin: &PackageOrigin,
+) -> Result<(), anyhow::Error> {
+    let mut manifest = get_manifest(reg, base_url, client, package, origin).await?;
+    println!("name: {}-{}", &package.name, &package.version);
+    println!("description: {}", &package.description);
+    let mut services: Vec<_> = manifest.services.into_iter().collect();
+    services.sort_by(|lhs, rhs| lhs.0.to_string().cmp(&rhs.0.to_string()));
+    manifest.data.sort_by(|lhs, rhs| {
+        (
+            lhs.account.to_string(),
+            lhs.service.to_string(),
+            &lhs.filename,
+        )
+            .cmp(&(
+                rhs.account.to_string(),
+                rhs.service.to_string(),
+                &rhs.filename,
+            ))
+    });
+
+    for account in &package.accounts {
+        let info = services
+            .binary_search_by(|service| service.0.to_string().cmp(&account.to_string()))
+            .map_or(None, |idx| Some(&services[idx].1));
+        print!(
+            "{}",
+            &ServicePrinter {
+                service: *account,
+                info,
+                data: get_service_data(&manifest.data, *account)
+            }
+        );
+    }
+    Ok(())
+}
+
+async fn package_info(
+    args: &Args,
+    mut client: reqwest::Client,
+    packages: &Vec<String>,
+    sources: &Vec<String>,
+) -> Result<(), anyhow::Error> {
+    let installed = PackageList::installed(&args.api, &mut client).await?;
```

Comment:
Also I know it's not part of this PR, but are they called apps or packages? `info` and `search` subcommand help text refers to "packages" but `install` and `list` refer to "apps"

Reply:
packages is probably better. I started using "apps" because apps represents what it means from the user's POV rather than from the tool's POV. However, I later realized a single app can have multiple packages for various reasons (optional components, size limits, etc), so "apps" isn't really correct.

Reply:
fixed